src/bitcoin-cli.cpp

```diff
@@ -291,6 +293,184 @@ class GetinfoRequestHandler: public BaseRequestHandler
+            const int mapped_as{peer["mapped_as"].isNull() ? 0 : peer["mapped_as"].get_int()};
+            const bool is_block_relay{!peer["relaytxes"].get_bool()};
+            const bool is_inbound{peer["inbound"].get_bool()};
+            m_conn_type conn_type{m_conn_type::ipv4};
+            if (is_inbound) {
+                if (IsAddrIPv6(addr)) {
+                    conn_type = m_conn_type::ipv6;
+                    ipv6_i += 1;
+                } else if (IsInboundOnion(mapped_as, addr, addr_local)) {
+                    conn_type = m_conn_type::onion;
+                    onion_i += 1;
+                } else {
+                    ipv4_i += 1;
+                }
+                if (is_block_relay) block_relay_i += 1;
+            } else {
+                if (IsAddrIPv6(addr)) {
+                    conn_type = m_conn_type::ipv6;
+                    ipv6_o += 1;
+                } else if (IsOutboundOnion(addr)) {
+                    conn_type = m_conn_type::onion;
+                    onion_o += 1;
+                } else {
+                    ipv4_o += 1;
+                }
+                if (is_block_relay) block_relay_o += 1;
+            }
+            if (m_verbose) {
+                // Push data for this peer to the peers vector.
+                const int peer_id{peer["id"].get_int()};
+                const int version{peer["version"].get_int()};
+                const std::string sub_version{peer["subver"].get_str()};
+                const int64_t conn_time{peer["conntime"].get_int64()};
+                const int64_t last_recv{peer["lastrecv"].get_int64()};
+                const int64_t last_send{peer["lastsend"].get_int64()};
+                const double min_ping{peer["minping"].isNull() ? 0 : peer["minping"].get_real()};
+                const double ping{peer["pingtime"].isNull() ? 0 : peer["pingtime"].get_real()};
+                peers.push_back({peer_id, mapped_as, version, conn_time, last_recv, last_send, min_ping, ping, addr, sub_version, conn_type, is_block_relay, !is_inbound});
+
+                is_asmap_on |= (mapped_as != 0);
+                max_peer_id_length = std::max(int(ToString(peer_id).length()), max_peer_id_length);
+                max_version_length = std::max(int((ToString(version) + sub_version).length()), max_version_length);
+            }
+        }
+        // Generate reports.
+        const UniValue& networkinfo{batch[ID_NETWORKINFO]["result"]};
+        std::string result{strprintf("%s %s - %i%s\n\n", PACKAGE_NAME, FormatFullVersion(), networkinfo["protocolversion"].get_int(), networkinfo["subversion"].get_str())};
+
+        // Report detailed peer connections list sorted by direction and minimum ping time.
+        if (m_verbose) {
+            std::sort(peers.begin(), peers.end());
+            result += "Peer connections sorted by direction and min ping\n<-> relay  conn minping   ping lastsend lastrecv uptime ";
+            if (is_asmap_on) result += " asmap ";
```

Comment:
Does it make sense to position this optional column as the last one? The "target audience" for this is humans, but maybe somebody will write a script that parses it and it will get bricked occasionally by the presence or absence of the `asmap` column if it is in the middle.

Also for humans - if somebody gets used to search for certain information at a certain position on the screen, e.g. `id`, it would be annoying if it gets shifted to the right by the occasional presence of an extra column.

Reply:
I gradually adjusted this over the past month of using it continually. Generally I don't think people change frequently between using asmap or not (based on my recent twitter usage polls, I suspect very few are using asmap yet). I initially wrote this using only asmap, and only realized that it would be annoying when @0xB10C mentioned it was just empty without it. When I did change to test it, it was better to remove it than have an empty column and wasted horizontal space. Thus, the current proposal. Feel free to propose a concrete change.

Reply:
> maybe somebody will write a script that parses it and it will get bricked occasionally by the presence or absence of the `asmap` column if it is in the middle.

Scripts should consume the RPC API, not the CLI.